### PR TITLE
Api microservice

### PR DESCRIPTION
**Pull request by EricLarson2020**
Connected obituary hub to microservice and built out covid and search by name endpoints with corresponding tests

### CODE REVIEW

File: app/controllers/obituaries/covid_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class Obituaries::CovidController < ApplicationController
   def index
-    @obituaries = Obituary.where(covid: true) 
+    @obituaries_api = ObituarySearch.new.search_covid_obituaries
```
Comment (EricLarson2020) [inline app/controllers/obituaries/covid_controller.rb:3]: The way I have this now both of these will be displayed on the screen, with the ones in our database on top and the ones from the api on the bottom

File: app/facades/obituary_search.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class ObituarySearch
```
Comment (sagemlee) [inline app/facades/obituary_search.rb:1]: Using the "facade"s makes me feel fancy

Comment (EricLarson2020) [inline app/facades/obituary_search.rb:1]: I will put all the search endpoints in here for obituaries

File: app/services/obituary_service.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
+class ObituaryService
+
```
Comment (EricLarson2020) [inline app/services/obituary_service.rb:2]: I will hit all the endpoints here, these are called from the search

File: app/views/layouts/_navigation.html.erb [hunk 4]
```diff
@@ -46,9 +46,9 @@
         <a class="nav-link" href="/obituaries/community">Community Obituaries</a>
       </li>
     </ul>
-    <form class="form-inline">
+    <form class="form-inline" id="search_form" action='/search' method="get">
```
Comment (EricLarson2020) [inline app/views/layouts/_navigation.html.erb:49]: edits to make form_button go places

File: app/views/obituaries/covid/index.html.erb [hunk 5]
```diff
@@ -13,4 +13,24 @@
   </div>
   <hr>
 <% end %>
+
+  <% @obituaries_api.each do |obit| %>
```
Comment (EricLarson2020) [inline app/views/obituaries/covid/index.html.erb:17]: These is where the styling comes in

File: app/controllers/obituaries/covid_controller.rb (hunk 1 above)
Comment (stellakunzang) [reply app/controllers/obituaries/covid_controller.rb:3]: excellent!

File: app/controllers/obituaries/covid_controller.rb [hunk 6]
```diff
@@ -1,5 +1,7 @@
...
+    @obituaries = Obituary.where(covid: true)
```
Comment (stellakunzang) [inline app/controllers/obituaries/covid_controller.rb:4]: I'm wondering if we want to add a sort to this, but that is probably better left to a future iteration!

File: app/controllers/search_controller.rb [hunk 7]
```diff
@@ -1,4 +1,6 @@
 class SearchController < ApplicationController
   def index
+
+      @obituaries_api = ObituarySearch.new.search_obituaries_by_name(params[:name])
```
Comment (stellakunzang) [inline app/controllers/search_controller.rb:4]: love those super clear class, method, and variable names!!

File: app/poros/api_obituary.rb [hunk 8]
```diff
@@ -0,0 +1,15 @@
+class ApiObituary
+  attr_reader :headline,
+              :abstract,
+              :web_url,
+              :pub_date,
+              :word_count
+              
+  def initialize(info)
+    @headline = info[:data][:attributes][:headline]
+    @abstract = info[:data][:attributes][:abstract]
```
Comment (stellakunzang) [inline app/poros/api_obituary.rb:10]: I'm curious about the abstract as opposed to the lead_paragraph and snippet that we get back from NYT. The data contained in these different attributes were sometimes identical and other times they varied. Did you explore this any further?

File: app/views/obituaries/covid/index.html.erb (hunk 5 above)
Comment (stellakunzang) [reply app/views/obituaries/covid/index.html.erb:17]: Nice! Maybe I can take this off your plate so you can finish building out search (if you aren't sick of it).

File: spec/features/obituaries/covid-19/index_spec.rb [hunk 9]
```diff
@@ -20,4 +20,35 @@
...
+      expect(web_url).not_to be_empty
+
+      expect(page).to have_css(".pub_date")
+      pub_date = find(".pub_date").text
+      expect(pub_date).not_to be_empty
+
+      expect(page).to have_css(".word_count")
+      word_count = find(".word_count").text
+      expect(word_count).not_to be_empty
+    end
```
Comment (stellakunzang) [inline spec/features/obituaries/covid-19/index_spec.rb:52]: great test!

File: app/services/obituary_service.rb [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+    resp = conn.get('/covid')
+    JSON.parse(resp.body, symbolize_names: true)
+  end
+
+  def get_obituaries_by_name(name)
+    resp = conn.get('/name') do |req|
+      req.params['name'] = "#{name}"
+    end
+    JSON.parse(resp.body, symbolize_names: true)
+  end
```
Comment (stellakunzang) [inline app/services/obituary_service.rb:17]: will this search for any name match? I'm not sure how the microservice is working. Maybe you and Jack can walk us through that this afternoon.

File: app/views/layouts/_navigation.html.erb (hunk 4 above)
Comment (stellakunzang) [reply app/views/layouts/_navigation.html.erb:49]: yay! The button finally does something!

File: app/facades/obituary_search.rb [hunk 11]
```diff
@@ -0,0 +1,17 @@
...
+  def search_covid_obituaries
+    json = ObituaryService.new.get_covid_obituaries
+    json.map do |obituary_data|
+      ApiObituary.new(obituary_data)
+    end
+  end
+
+  def search_obituaries_by_name(name)
+    json = ObituaryService.new.get_obituaries_by_name(name)
+    json.map do |obituary_data|
```
Comment (jpc20) [inline app/facades/obituary_search.rb:12]: Nice encapsulation.

File: app/services/obituary_service.rb [hunk 12]
```diff
@@ -0,0 +1,26 @@
...
+
+  def get_covid_obituaries
+    resp = conn.get('/covid')
+    JSON.parse(resp.body, symbolize_names: true)
+  end
+
+  def get_obituaries_by_name(name)
+    resp = conn.get('/name') do |req|
+      req.params['name'] = "#{name}"
```
Comment (jpc20) [inline app/services/obituary_service.rb:14]: Cool. I am excited to see what type of results we get.

File: app/services/obituary_service.rb (hunk 10 above)
Comment (sagemlee) [reply app/services/obituary_service.rb:17]: Good question Stella. Curious to know if it only takes letters, if it sterilizes special characters, also how it does with spaces?

**Pull request merged**